### PR TITLE
refactor_getPersonInformation_method_and_person_controller_integratio…

### PR DESCRIPTION
**Pull request by G-jonathan**
Sorry it's a little untidy but the most important to checked is i think:
/integration/MedicalRecordControllerTestIt.java

EDIT- force-push n°1 -->  Mainly a new file unitTest: **MedicalRecordServiceTest.java**

EDIT- force-push n°2 --> Add **PersonServiceTest** 

EDIT- force-push n°3 --> Add **FireStationServiceTest**

### CODE REVIEW

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/controller/MedicalRecordController.java [hunk 1]
```diff
@@ -40,12 +40,15 @@
         String functionPath = CLASSPATH + ".addMedicalRecord";
         log.info("Request received  in " + functionPath);
         try {
-            medicalRecordService.addMedicalRecord(medicalRecord);
+           boolean response = medicalRecordService.addMedicalRecord(medicalRecord);
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/controller/MedicalRecordController.java:43]: indentation seems weird there

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java [hunk 2]
```diff
@@ -104,16 +104,23 @@ public boolean deleteAPerson(UniqueIdentifier uniqueIdentifier) {
...
         Optional<MedicalRecord> medicalRecord = medicalRecordRepository.findMedicalRecordByFirstNameAndLastName(firstName, lastName);
-        AgeCalculationModel ageCalculationModel = new AgeCalculationModel(medicalRecord.get().getBirthdate(), "MM/dd/yyyy");
-        String age = String.valueOf(AgeCalculator.ageCalculator(ageCalculationModel));
         PersonInformation personInformation = new PersonInformation();
+        if (medicalRecord.isPresent()){
+            AgeCalculationModel ageCalculationModel = new AgeCalculationModel(medicalRecord.get().getBirthdate(), "MM/dd/yyyy");
+            personInformation.setAge(String.valueOf(AgeCalculator.ageCalculator(ageCalculationModel)));
+            personInformation.setMedications(medicalRecord.get().getMedications());
+            personInformation.setAllergies(medicalRecord.get().getAllergies());
+        }
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java:113]: else should be at this level
```
} else {
...
}
```

**Review by Asavarkhul — APPROVED** [review]

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/controller/MedicalRecordController.java (hunk 1 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/controller/MedicalRecordController.java:43]: done

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java (hunk 2 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java:113]: done

**Pull request merged**